test/framework/toolchain.py

```diff
@@ -95,23 +96,69 @@ def test_unknown_toolchain(self):
 
+    def test_system_toolchain(self):
+        """Test for system toolchain."""
+        for ver in ['system', '']:
+            tc = self.get_toolchain('system', version=ver)
+            self.assertTrue(isinstance(tc, SystemToolchain))
+
     def test_foss_toolchain(self):
         """Test for foss toolchain."""
         self.get_toolchain("foss", version="2018a")
 
-    def test_get_variable_dummy_toolchain(self):
-        """Test get_variable on dummy toolchain"""
-        tc = self.get_toolchain('dummy', version='dummy')
-        tc.prepare()
-        self.assertEqual(tc.get_variable('CC'), '')
-        self.assertEqual(tc.get_variable('CXX', typ=str), '')
-        self.assertEqual(tc.get_variable('CFLAGS', typ=list), [])
+    def test_get_variable_system_toolchain(self):
+        """Test get_variable on system/dummy toolchain"""
 
-        tc = self.get_toolchain('dummy', version='')
-        tc.prepare()
-        self.assertEqual(tc.get_variable('CC'), '')
-        self.assertEqual(tc.get_variable('CXX', typ=str), '')
-        self.assertEqual(tc.get_variable('CFLAGS', typ=list), [])
+        # system toolchain version doesn't really matter, but fine...
+        for ver in ['system', '']:
+            tc = self.get_toolchain('system', version=ver)
+            tc.prepare()
+            self.assertEqual(tc.get_variable('CC'), '')
+            self.assertEqual(tc.get_variable('CXX', typ=str), '')
+            self.assertEqual(tc.get_variable('CFLAGS', typ=list), [])
+
+        # dummy toolchain is deprecated, so we need to allow for it (and catch the warnings that get printed)
+        self.allow_deprecated_behaviour()
+
+        for ver in ['dummy', '']:
+            self.mock_stderr(True)
+            tc = self.get_toolchain('dummy', version=ver)
+            self.mock_stderr(False)
+            tc.prepare()
+            self.assertEqual(tc.get_variable('CC'), '')
+            self.assertEqual(tc.get_variable('CXX', typ=str), '')
+            self.assertEqual(tc.get_variable('CFLAGS', typ=list), [])
+
+    def test_is_system_toolchain(self):
+        """Test is_system_toolchain method."""
+        for ver in ['system', '']:
+            tc = self.get_toolchain('system', version=ver)
+            self.assertTrue(tc.is_system_toolchain())
+
+        tc = self.get_toolchain('foss', version='2018a')
+        self.assertFalse(tc.is_system_toolchain())
+
+        self.setup_sandbox_for_intel_fftw(self.test_prefix)
+        self.modtool.prepend_module_path(self.test_prefix)
+        tc = self.get_toolchain('intel', version='2018a')
+        self.assertFalse(tc.is_system_toolchain())
+
+        # using dummy toolchain is deprecated, so to test for that we need to explicitely allow using deprecated stuff
+        error_pattern = "Use of 'dummy' toolchain is deprecated"
+        for ver in ['dummy', '']:
+            self.assertErrorRegex(EasyBuildError, error_pattern, self.get_toolchain, 'dummy', version=ver)
```

Comment:
Is that line correct?

Reply:
What would be wrong with it?

By default, the tests run in a setup where triggering deprecated stuff results in an error being raised, which is exactly what we're checking here.

Reply:
So, with EasyBuild 4.0, you'll get:

```
$ eb bzip2-1.0.6.eb
== temporary log file in case of crash /tmp/eb-Oxg9IR/easybuild-RZK2aJ.log

WARNING: Deprecated functionality, will no longer work in v5.0: Use of 'dummy' toolchain is deprecated, use 'system' toolchain instead; see http://easybuild.readthedocs.org/en/latest/Deprecated-functionality.html for more information

== processing EasyBuild easyconfig /easybuild-easyconfigs/easybuild/easyconfigs/b/bzip2/bzip2-1.0.6.eb
...
```

But with EasyBuild 5.0 and newer, this will result in an error, which you can already mimic using `--deprecated` (the tests basically run with `--deprecated=100000` unless told otherwise to signal unintended triggering of deprecated functionality)

```
$ eb bzip2-1.0.6.eb --deprecated=5.0
== temporary log file in case of crash /tmp/eb-7bD941/easybuild-x66BGe.log
ERROR: Failed to process easyconfig /easybuild-easyconfigs/easybuild/easyconfigs/b/bzip2/bzip2-1.0.6.eb: DEPRECATED (since v5.0) functionality used: Use of 'dummy' toolchain is deprecated, use 'system' toolchain instead; see http://easybuild.readthedocs.org/en/latest/Deprecated-functionality.html for more information
```

Reply:
I didn't bother looking at what assertErrorRegex takes for argument and expected self.get_toolchain to be a function call, not the function pointer it apparently is...